src/Sentry/Laravel/Tracing/Storage/TracingFilesystem.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+namespace Sentry\Laravel\Tracing\Storage;
+
+use Exception;
+use Illuminate\Contracts\Filesystem\Cloud as CloudFilesystemContract;
+use Illuminate\Contracts\Filesystem\Filesystem;
+use Illuminate\Filesystem\FilesystemAdapter;
+use Sentry\Tracing\SpanContext;
+use function Sentry\trace;
+
+/**
+ * Decorates the underlying filesystem by wrapping all calls to it with tracing.
+ *
+ * Parameters such as paths, directories or options are attached to the span as data,
+ * parameters that contain file contents are omitted due to potential problems with
+ * payload size or sensitive data.
+ */
+class TracingFilesystem implements CloudFilesystemContract
```

Comment:
I am wondering about this, we might want to have a seperate `TracingCloudFilesystem` and a `TracingFilesystem` calls. People might test for the implementing interface to see how a file system can be used.

Reply:
Laravel always implements `Cloud` and never just its parent, so I think we should be fine to do the same:

![image](https://github.com/getsentry/sentry-laravel/assets/12158000/633c81ad-6097-43cd-b46a-d62bb5fe3024)

![image](https://github.com/getsentry/sentry-laravel/assets/12158000/0fc95e42-2651-41d5-bd36-6a0ead40013f)

Reply:
Those packages seem fundamentally flawed in their assumption that this distinction is meaningful though, given that Laravel defaults to wrapping even local filesystems in `FilesystemAdapter` - which does implement `Cloud`.

`FilesystemAdapter` tries really hard to return a value for `url()`, even for local filesystems:

![image](https://github.com/getsentry/sentry-laravel/assets/12158000/d550c0e2-e494-4b93-98ba-08e0eb25c9dc)

I think that issues with this would be very unlikely, but I am working on a solution to separate the implemenations.

Reply:
Done with [`f40bbbf` (#726)](https://github.com/getsentry/sentry-laravel/pull/726/commits/f40bbbf4bc92ad41c4794a9b7a4d797d23ddb10a).